### PR TITLE
Adopt conditional code loading from solidus_support

### PR DESCRIPTION
**Pull request by aldesantis**
Adopts the `SolidusSupport::EngineExtensions` module for loading code conditionally depending on which Solidus engines are available. This is mostly a DRYed up version of what was already in this gem.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -24,6 +24,8 @@ gem 'rails-controller-testing', group: :test
 
 gemspec
 
+gem 'solidus_support', github: 'solidusio/solidus_support'
```
Comment (kennyadsl) [inline Gemfile]: I've just released `solidus_support`0.5.0 with `SolidusSupport::EngineExtensions`.  We should be able to remove this line now.

Comment (kennyadsl) [reply Gemfile]: Hey @aldesantis, when you have some time, can you please remove this line so we'll be able to merge this?

**Comment by aldesantis** [comment]
@kennyadsl removed the GitHub reference to solidus_support and rebased.

**Comment by kennyadsl** [comment]
@aldesantis please rebase again, I merged the PR that was breaking specs. 🙏

**Comment by aldesantis** [comment]
@kennyadsl done!

**Pull request merged**